docs/dev/best_practices.md

```diff
@@ -310,3 +310,17 @@ Next, you need to put the following jar files into the `lib/` folder:
 Note that you need to explicitly set the `lib/` directory when using a per job YARN cluster.
 
 The command to submit Flink on YARN with a custom logger is: `./bin/flink run -yt $FLINK_HOME/lib <... remaining arguments ...>`
+
+
+## Resolving Dependency Conflicts with Flink using the maven-shade-plugin.
```

Comment:
I think this is better for the debugging section

https://ci.apache.org/projects/flink/flink-docs-release-1.3/monitoring/debugging_classloading.html